spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/test/java/org/springframework/cloud/stream/binder/redis/RedisBinderTests.java

```diff
@@ -145,16 +137,16 @@ public void testProducerProperties() throws Exception {
 		properties.put(BinderPropertyKeys.NEXT_MODULE_COUNT, "1");
 
 		producerBinding = binder.bindProducer("props.0", new DirectChannel(), properties);
-		assertEquals(2, bindings.size());
-		endpoint = bindings.get(1).getEndpoint();
+		endpoint = extractEndpoint(producerBinding);
 		adapter = (RedisQueueOutboundChannelAdapter) TestUtils.getPropertyValue(endpoint, "handler.adapters", Map.class).get("test");
 		assertEquals(
 				"'props.0.test-' + headers['partition']",
 				TestUtils.getPropertyValue(adapter, "queueNameExpression", Expression.class).getExpressionString());
 
-		binder.unbind(producerBinding);
-		binder.unbind(consumerBinding);
-		assertEquals(0, bindings.size());
+		producerBinding.unbind();
+		consumerBinding.unbind();
+		assertFalse(extractEndpoint(producerBinding).isRunning());
```

Comment:
already have `endpoint` assigned above (but should rename to `producerEndpoint` for clarity)